tests/e2e/test_reborn_capability_policy_xyzorg.py

```
@@ -0,0 +1,975 @@
+async def _create_thread(client, base_url, token):
+    resp = await client.post(
+        f"{base_url}/api/webchat/v2/threads",
+        headers=_bearer(token),
+        json={"client_action_id": str(uuid.uuid4())},
+        timeout=15,
+    )
+    resp.raise_for_status()
+    return resp.json()["thread"]["thread_id"]
+
+
+async def _offered_tool_names(client, mock_llm_server):
```

Ruff (0.15.18)

[warning] Missing return type annotation for private function `_offered_tool_names`

(ANN202)